### DISCUSSION TITLE
Occasionally substitutes 'localhost' in place of $DOMAIN set in .env file

### DISCUSSION THREAD

**Discussion by benfishbus** [dani-garcia/vaultwarden#5156] [Q&A]
I have two-step login enabled using Duo. When I try to log in from a new browser - either the browser extension or the web vault - I am able to trigger and acknowledge the Duo push, then the browser session tries to go here: 

`http://localhost/duo-redirect-connector.html?client=web&state={string1}&code={string2}
`

If I quickly replace 'http://localhost' with my $DOMAIN prefix, it finishes authenticating. Otherwise I'm dead in the water.

I think this started happening a few months ago. I access the official vw docker container through a cloudflared tunnel via public hostname with no additional origin configurations.

This might be a problem with my Duo or cloudflared setup rather than with vw itself. Anyone else experiencing such problems? I see a few really old threads, but nothing too recent.

**Comment by BlackDex** [comment]
See https://github.com/dani-garcia/vaultwarden/wiki/Configuration-overview#setting-the-domain-url

Make sure you have Vaultwarden correctly configured.
